tests/karma.js

```diff
@@ -2156,3 +2277,50 @@ beforeAll(function (done) {
 afterAll(function (done) {
   provider.finalize().then(done, done.fail)
 })
+
+function buildPropfindFileList (token) {
+  let contentResponse = ''
+  for (let fileNum = 0; fileNum < config.testFiles.length; fileNum++) {
+    contentResponse +=
+      '<d:response>\n' +
+      // TODO add ${SUBFOLDER} in the href
+      '   <d:href>/remote.php/dav/public-files/' + token + '/' +
+                  encodeURIComponent(config.testFiles[fileNum]) + '</d:href>\n' +
+      '   <d:propstat>\n' +
+      '      <d:prop>\n' +
+      '         <d:getcontenttype>text/plain</d:getcontenttype>\n' +
+      '      </d:prop>\n' +
+      '      <d:status>HTTP/1.1 200 OK</d:status>\n' +
+      '   </d:propstat>\n' +
+      '   <d:propstat>\n' +
+      '      <d:prop>\n' +
+      '         <oc:public-link-item-type />\n' +
+      '         <oc:public-link-permission />\n' +
+      '         <oc:public-link-expiration />\n' +
+      '         <oc:public-link-share-datetime />\n' +
+      '         <oc:public-link-share-owner />\n' +
+      '      </d:prop>\n' +
+      '      <d:status>HTTP/1.1 404 Not Found</d:status>\n' +
+      '   </d:propstat>\n' +
+      '</d:response>\n'
+  }
+  return contentResponse
+}
+
+function propfindBody (token) {
```

Comment:
This function returns a propfind body for a public-files request.
So either:
1) it will become more flexible in future, and take more parameters, and become a general `propfindBody` function, or;
2) its name could be changed to something like `propfindBodyForPublicFiles` or...

But that can be discussed in a upcoming refactor discussion.